src/field/field255/integer.rs

```diff
@@ -0,0 +1,190 @@
+    }
+}
+
+impl Add<Field255Integer> for Field255Integer {
+    type Output = Field255Integer;
+
+    fn add(self, rhs: Field255Integer) -> Self::Output {
+        Field255Integer(self.0 + rhs.0)
+    }
+}
+
+impl Sub<Field255Integer> for Field255Integer {
+    type Output = Field255Integer;
+
+    fn sub(self, rhs: Field255Integer) -> Self::Output {
+        Field255Integer(self.0 - rhs.0)
+    }
+}
+
+impl Mul<Field255Integer> for Field255Integer {
+    type Output = Field255Integer;
+
+    fn mul(self, rhs: Field255Integer) -> Self::Output {
+        Field255Integer(self.0 * rhs.0)
+    }
+}
+
+impl Div<Field255Integer> for Field255Integer {
+    type Output = Field255Integer;
+
+    fn div(self, rhs: Field255Integer) -> Self::Output {
+        Field255Integer(self.0 / rhs.0)
+    }
+}
+
+impl Rem<Field255Integer> for Field255Integer {
+    type Output = Field255Integer;
+
+    fn rem(self, rhs: Field255Integer) -> Self::Output {
+        Field255Integer(self.0 % rhs.0)
+    }
+}
+
+impl Shl for Field255Integer {
+    type Output = Field255Integer;
+
+    fn shl(self, rhs: Self) -> Self::Output {
+        Field255Integer(self.0 << u128::try_from(rhs.0).unwrap())
+    }
+}
+
+impl Shr for Field255Integer {
+    type Output = Field255Integer;
+
+    fn shr(self, rhs: Self) -> Self::Output {
+        Field255Integer(self.0 >> u128::try_from(rhs.0).unwrap())
+    }
+}
+
+impl BitAnd for Field255Integer {
+    type Output = Field255Integer;
+
+    fn bitand(self, rhs: Self) -> Self::Output {
+        Field255Integer(self.0 & rhs.0)
+    }
+}
+
+impl Debug for Field255Integer {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        <BigUint as Debug>::fmt(&self.0, f)
+    }
+}
+
+impl Display for Field255Integer {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        <BigUint as Display>::fmt(&self.0, f)
+    }
+}
+
+impl From<usize> for Field255Integer {
+    fn from(value: usize) -> Self {
+        Field255Integer(value.into())
+    }
+}
+
+impl TryFrom<Field255Integer> for u64 {
+    type Error = TryFromBigIntError<BigUint>;
+
+    fn try_from(value: Field255Integer) -> Result<Self, Self::Error> {
+        u64::try_from(value.0)
+    }
+}
+
+impl From<Field255Integer> for Field255 {
+    fn from(value: Field255Integer) -> Self {
+        let le_bytes_vec = value.0.rem(&MODULUS.0).to_bytes_le();
+
+        let mut le_bytes_array = [0u8; 32];
+        le_bytes_array[..le_bytes_vec.len()].copy_from_slice(&le_bytes_vec);
+
+        let mut output = [0; 5];
+        fiat_25519_from_bytes(&mut output, &le_bytes_array);
+        Field255(output)
+    }
+}
+
+impl From<Field255> for Field255Integer {
+    fn from(value: Field255) -> Self {
+        let mut le_bytes = [0u8; 32];
+        fiat_25519_to_bytes(&mut le_bytes, &value.0);
+        Field255Integer::from_little_endian(&le_bytes)
+    }
+}
+
+impl TestFieldElementWithInteger for Field255 {
+    type Integer = Field255Integer;
```

Comment:
It would be preferable to use the `BigUint` type directly so that we don't have to define a wrapper type. Playing with this a little bit, it looks like the only thing thing that `BigUint` doesn't have that we need is bit shifting by `BigUint`, i.e., `BigUint << BigUint`. Howver, `BigUint` does implement bit shifting for various primitive types.

My expectation is that we could just require bit shift by a primitive type, e.g., `u64`. I can implement this in a follow-up PR, if you prefer to not have to change things.

```suggestion
    type Integer = BigUint;
```

Reply:
Yes, this sounds good, it looks like all the primitives support shifting by any other primitive. From a quick glance, it looks like most `F::Integers` constructed for use in shifting come from `usize` be they in the bitvector representation helper methods or `FieldParameters.num_roots`.